tools/test/travis-ci/scancode-evaluate.py

```diff
@@ -0,0 +1,105 @@
+"""
+SPDX-License-Identifier: Apache-2.0
+
+Copyright (c) 2020 Arm Limited. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations 
+"""
+
+# Asumptions for this script: 
+# 1. directory_name is scanned directory.
+#  Files are copied to this directory with full tree. As result, if we find
+#  license offender, we can have full path (just scrape directory_name). We do this
+#  magic because scancode allows to scan directories/one file.
+# 2. SPDX and license text is a must for all code files
+
+import json
+import argparse
+import sys
+import os.path
+
+MISSING_LICENSE_TEXT = "Missing license header"
+MISSING_PERMISIVE_LICENSE_TEXT = "Non-permissive license"
+MISSING_SPDX_TEXT = "Missing SPDX license identifier"
+
+FILE_OFFENDER_TEMPLATE = {
+   'file': None,
+   'reason': None
+}
+
+def license_check(directory_name, file):
+
+    offenders = []
+
+    # find all licenses in the files, must be licensed and permissive
+    with open(file, 'r') as scancode_output:
+        results = json.load(scancode_output)
```

Comment:
Could this fail to load the json if the json is invalid? ie should there by a try around the json.load?

Reply:
Fixed, via ValueError catch